hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieFileSliceReader.java

```diff
@@ -41,11 +43,26 @@
     while (baseIterator.hasNext()) {
       GenericRecord record = (GenericRecord)  baseIterator.next();
       HoodieRecord<T> hoodieRecord = SpillableMapUtils.convertToHoodieRecordPayload(record, payloadClass);
-      scanner.processNextRecord(hoodieRecord);
+      processNextRecord(scanner, hoodieRecord);
     }
     return new HoodieFileSliceReader(scanner.iterator());
   }
 
+  private static void processNextRecord(HoodieMergedLogRecordScanner scanner, HoodieRecord<? extends HoodieRecordPayload> hoodieRecord) {
```

Comment:
IIUC you write commit 1 + commit2 + (commit3 with an inline clustering), but the clustering only read commit 1 + commit 2. Is that correct? How does this solve the problem?

Reply:
thanks for your review.    commit3 is only to use trigger clustering.   
in the commit1 , we create a table and the age column is assigned to 1
in the commit2，update age value to 1001 where keyid < 5 to produce log files
in the commit3，only use to trigger clustering and find the bug。

In the previous code， we combine base file record and log file record by call HoodieMergedLogRecordeScanner.processNextRecord, however this function is not suitable for this scene，if we call this function，the record from base file will be kept， and the record from log file will be discard，this is wrong。

Reply:
We should always to call `preCombine` or `combineAndGetUpdateValue` to determine which record to keep. I see `processNextRecord` is calling preCombine but still using the Deprecated API. If we update it to `preCombine(T another, Properties properties)`, the order should be expected then?

Reply:
I'm afraid it will lead some complex changes
1）all implementation classes for HoodieRecordPayload has not implemented  preCombine(T another, Properties properties) 
2)  we need modfiy preCombine(T another, Properties properties)  to preCombine(T another, Properties properties, Scheme schema), since we need to get the value of  HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP from current payload to do compare with another payload.   Obviously, we can pass in a special value to determine which value to keep， but i think this is a tricky。

of course， If you insist， i will update code

Reply:
let's discuss before we move on. @nsivabalan thoughts?